Source/Tools/OHTransfer/OHTransfer.cs

```diff
@@ -0,0 +1,654 @@
+                else
+                    signalReferenceIDMap[(destinationSignal.Kind, destinationSignal.Index)] = destinationEntry.Key;
+            }
+
+            foreach (KeyValuePair<ulong, SignalReference> sourceEntry in sourceCSVData)
+            {
+                SignalReference sourceSignal = sourceEntry.Value;
+
+                if (sourceSignal.Kind == SignalKind.Calculation)
+                {
+                    if (calculatedSignalReferenceIDMap.TryGetValue((sourceSignal.Kind, sourceSignal.Index, sourceSignal.Acronym ?? string.Empty), out ulong destID))
+                        historianIDMapping[sourceEntry.Key] = destID;
+                }
+                else
+                {
+                    if (signalReferenceIDMap.TryGetValue((sourceSignal.Kind, sourceSignal.Index), out ulong destID))
+                        historianIDMapping[sourceEntry.Key] = destID;
+                }
+            }
+
+            if (historianIDMapping.Count == 0)
+            {
+                ShowUpdateMessage("No matching measurements found between source and destination CSV files -- operation cancelled.");
+                return;
+            }
+
+            ShowUpdateMessage("Found {0:N0} matching measurements between source and destination CSV files.", historianIDMapping.Count);
+
+            string sourcePath = textBoxSourceFiles.Text.Trim();
+            string destinationPath = textBoxDestinationFiles.Text.Trim();
+
+            m_cancellationTokenSource = new CancellationTokenSource();
+
+            new Thread(() =>
+            {
+                try
+                {
+                    EnableGoButton(false);
+
+                    CancellationToken cancellationToken = m_cancellationTokenSource.Token;
+                    
+                    int totalDays = (int)(endTime - startTime).TotalDays;
+                    SetProgressMaximum(totalDays);
+
+                    const string InstanceName = "PPA";
+
+                    // Establish archive information for this historian instance
+                    HistorianServerDatabaseConfig archiveInfo = new(InstanceName, sourcePath, false)
+                    {
+                        TargetFileSize = (long)(1.5D * SI.Giga),
+                        DirectoryMethod = ArchiveDirectoryMethod.YearThenMonth
+                    };
+
+                    HistorianServer server = new(archiveInfo);
+                    SnapClient client = SnapClient.Connect(server.Host);
+                    ClientDatabaseBase<HistorianKey, HistorianValue> database = client.GetDatabase<HistorianKey, HistorianValue>(InstanceName);
+                    Dictionary<Guid, ArchiveDetails> attachedFiles = database.GetAllAttachedFiles()
+                        .OrderBy(file => file.EndTime)
+                        .ToDictionary(file => file.Id, file => file);
```

Comment:
What does the `OrderBy()` do here? The dictionary is an unordered collection. Also, the value selector should be unnecessary.
```suggestion
                    Dictionary<Guid, ArchiveDetails> attachedFiles = database.GetAllAttachedFiles()
                        .ToDictionary(file => file.Id);
```